includes/data-stores/class-wc-admin-reports-orders-data-store.php

```diff
@@ -173,50 +176,84 @@ protected function orders_stats_sql_filter( $query_args, &$totals_query, &$inter
 		// TODO: performance of all of this?
 		global $wpdb;
 
-		$where_clause = '';
-		$from_clause  = '';
-
+		$from_clause        = '';
 		$orders_stats_table = $wpdb->prefix . self::TABLE_NAME;
-		$allowed_products   = $this->get_allowed_products( $query_args );
+		$operator           = $this->get_match_operator( $query_args );
 
-		if ( count( $allowed_products ) > 0 ) {
-			$allowed_products_str = implode( ',', $allowed_products );
+		$where_filters = array();
 
-			$where_clause .= " AND {$orders_stats_table}.order_id IN (
+		// TODO: maybe move the sql inside the get_included/excluded functions?
+		// Products filters.
+		$included_products = $this->get_included_products( $query_args );
+		$excluded_products = $this->get_excluded_products( $query_args );
+		if ( $included_products ) {
+			$where_filters[] = " {$orders_stats_table}.order_id IN (
 			SELECT
 				DISTINCT {$wpdb->prefix}wc_order_product_lookup.order_id
 			FROM
 				{$wpdb->prefix}wc_order_product_lookup
 			WHERE
-				{$wpdb->prefix}wc_order_product_lookup.product_id IN ({$allowed_products_str})
+				{$wpdb->prefix}wc_order_product_lookup.product_id IN ({$included_products})
 			)";
 		}
 
-		if ( is_array( $query_args['coupons'] ) && count( $query_args['coupons'] ) > 0 ) {
-			$allowed_coupons_str = implode( ', ', $query_args['coupons'] );
-
-			$where_clause .= " AND {$orders_stats_table}.order_id IN (
+		if ( $excluded_products ) {
```

Comment:
Looks like you can combine this if/else into one

Reply:
Not sure what you mean? There is no else there?

Reply:
Oops, yes. I meant the if statement above is nearly identical except for `$excluded_products`/`$included_products`.